oneflow/user/kernels/elementwise_maximum_minimum_kernel_util.h

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2020 The OneFlow Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#ifndef _ONEFLOW_USER_KERNEL_ELEMENTWISE_MAXIMUM_MINIMUM_KERNEL_UTIL_H
+#define _ONEFLOW_USER_KERNEL_ELEMENTWISE_MAXIMUM_MINIMUM_KERNEL_UTIL_H
+#include "oneflow/core/common/data_type.h"
+#include "oneflow/core/ndarray/xpu_util.h"
+#include "oneflow/core/framework/framework.h"
+#include "oneflow/core/kernel/kernel_util.h"
+
+namespace oneflow {
+namespace user_op {
+
+template<typename T>
+struct MaximumUtil {
+  OF_DEVICE_FUNC T operator()(T x, T y) const { return x > y ? x : y; }
+
+  OF_DEVICE_FUNC static void Backward(const T* dz, const T* x, const T* y, T* dx, T* dy) {
+    if (*x > *y) {
+      if (dx) { *dx = *dz; }
+      if (dy) { *dy = 0; }
+    } else {
+      if (dy) { *dy = *dz; }
+      if (dx) { *dx = 0; }
+    }
```

Comment:
```suggestion
    const T dz_val = *dz;
    T dx_val = 0;
    T dy_val = 0;
    if (*x > *y) {
      dx_val  = dz_val;
    } else {
      dy_val = dz_val;
    }
    if (dx) { *dx = dx_val; }
    if (dy) { *dy = dy_val; }
```
这样处理性能会更好，一般情况下，应避免分支里面的 global memory 访问

Reply:
学习了